owners/src/repo/abstract_repo.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2019 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Interface for reading from a GitHub repository.
+ */
+module.exports = class AbstractRepository {
```

Comment:
Some suggestions:
- Rename the base class just `Repository`
- Add `@interface` to the JSDoc to indicate that this is an abstract class
- Implement a placeholder constructor that throws if someone tries to instantiate the base class

Reply:
Re: first 2, done.
Re: your third point, that won't work out. ES6 constructors have to call their super constructor, so the constructor can't throw. Adding the interface JSDoc though.

Reply:
Might something like this work?
```js
constructor() {
  if (this.constructor === Repository) {
    throw new TypeError('Cannot instantiate base class'); 
  }
}
```

Reply:
See #568